ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcService.java

```diff
@@ -231,4 +323,40 @@ public StartLeaderElectionReplyProto startLeaderElection(StartLeaderElectionRequ
     final RaftPeerId target = RaftPeerId.valueOf(request.getServerRequest().getReplyId());
     return getProxies().getProxy(target).startLeaderElection(request);
   }
+
+  private static final class ServerConfig {
+
+    private final int port;
+    private final GrpcTlsConfig tlsConfig;
+
+    private ServerConfig(int port, GrpcTlsConfig tlsConfig) {
+      this.port = port;
+      this.tlsConfig = tlsConfig;
+    }
+
+    public int getPort() {
+      return port;
+    }
+
+    public GrpcTlsConfig getTlsConfig() {
+      return tlsConfig;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      ServerConfig that = (ServerConfig) o;
+      return port == that.port && port > 0;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(port);
+    }
```

Comment:
We may compare getPort() directly as mentioned above.  Then, we don't have to override equals(..) and hashCode() here.